### PR TITLE
[MLv2] Support segments

### LINKED ISSUE #32409: [MLv2] Migrate legacy segments to MLv2
MLv1 allows to apply a legacy segment as a filter. We should follow a similar route we went with legacy metrics for MLv2 aggregations (#31871)

- [ ] `available-segments` method
- [ ] `display-info` implementation for segments (should have a display name and a description)
- [ ] if a segment is used in a filter clause, it should be marked as selected in the `available-segments` list or something
- [ ] there should be a way to apply a segment filter. E.g. `filter` method could accept a segment object
- [ ] it should be possible to replace one filter segment with another using `replace-clause` (similar to #32049)

### PR DESCRIPTION
**Pull request by metamben**
Fixes #32409.

Advice about handling named segment references is welcome.

### CODE REVIEW

File: test/metabase/lib/schema/filter_test.cljc [hunk 1]
```diff
@@ -71,8 +71,7 @@
            [:time-interval field :last :hour]
            [:time-interval field 4 :hour]
            [:time-interval {:include-current true} field :next :day]
-           [:segment 1]
-           [:segment "segment-id"]]]
```
Comment (snoe) [inline test/metabase/lib/schema/filter_test.cljc:75]: Why remove these tests?

File: test/metabase/lib/segment_test.cljc [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+  (testing "Include `:selected false` in display info for Segments not in aggregations"
+    (are [segment] (not (:selected (lib.metadata.calculation/display-info lib.tu/venues-query segment)))
+      segment-clause
+      segment-metadata)))
+
+(deftest ^:parallel unknown-display-info-test
+  (is (=? {:effective-type    :type/Boolean
+           :display-name      "[Unknown Segment]"
+           :long-display-name "[Unknown Segment]"}
+          (lib.metadata.calculation/display-info query-with-segment [:segment {} (inc segment-id)]))))
```
Comment (snoe) [inline test/metabase/lib/segment_test.cljc:69]: should test named segments here too I believe.

File: src/metabase/lib/segment.cljc [hunk 3]
```diff
@@ -1,18 +1,79 @@
...
+   [metabase.lib.options :as lib.options]
+   [metabase.lib.ref :as lib.ref]
+   [metabase.lib.schema :as lib.schema]
+   [metabase.lib.schema.id :as lib.schema.id]
+   [metabase.lib.util :as lib.util]
+   [metabase.shared.util.i18n :as i18n]
+   [metabase.util.malli :as mu]))
+
+(defn- resolve-segment [query segment-id]
+  (when (integer? segment-id)
```
Comment (snoe) [inline src/metabase/lib/segment.cljc:16]: Are named segments irrecoverable? I see this is the same as metrics.

**Review by snoe — APPROVED** [review]
LGTM

**Review by camsaul — APPROVED** [review]

File: src/metabase/lib/segment.cljc (hunk 3 above)
Comment (metamben) [reply src/metabase/lib/segment.cljc:16]: Yes, I kept to the metrics way.

**Pull request merged**